graphdatascience/query_runner/arrow_query_runner.py

```diff
@@ -28,16 +31,15 @@ def __init__(
             else flight.Location.for_grpc_tcp(host, int(port_string))
         )
 
-        self._flight_client = flight.FlightClient(location, disable_server_verification=disable_server_verification)
-        self._flight_options = flight.FlightCallOptions()
-
+        client_options: Dict[str, Any] = {"disable_server_verification": disable_server_verification}
         if auth:
-            username, password = auth
-            header, token = self._flight_client.authenticate_basic_token(username, password)
-            if header:
-                self._flight_options = flight.FlightCallOptions(headers=[(header, token)])
+            client_options["middleware"] = [AuthFactory(auth)]
+
+        self._flight_client = flight.FlightClient(location, **client_options)
 
-    def run_query(self, query: str, params: Dict[str, Any] = {}) -> DataFrame:
+    def run_query(self, query: str, params: Optional[Dict[str, Any]] = None) -> DataFrame:
```

Comment:
I don't understand why we changed this interface. Now we have these if statements everywhere. Would it be possible to change that back?

Reply:
@knutwalker (as well as Intellij) remarked on this a a potential bug. The default dict that is created is a shared instance through the lifetime of the class instance/interpreter (?). If it is changed at some point these changes would be seen by any invocation of the method.

Reply:
Ah, good catch!